### PR TITLE
Substack importer: add translation functions to stable strings

### PR DESCRIPTION
**Pull request by simison**
## Proposed Changes

* Wraps most of the string with translation functions. The UI itself is still behind feature flags so fine to merge to prod.

Some of the screens:

Screenshot 2024-10-10 at 18 04 34
Screenshot 2024-10-10 at 18 07 15
Screenshot 2024-10-10 at 18 06 03
Screenshot 2024-10-10 at 18 05 37
Screenshot 2024-10-10 at 18 05 25
Screenshot 2024-10-10 at 18 05 15
Screenshot 2024-10-10 at 18 05 03
Screenshot 2024-10-10 at 18 04 44

## Why are these changes being made?

*

## Testing Instructions

* Test the flow from Tools -> Import -> Substack

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx [hunk 1]
```diff
@@ -88,7 +86,17 @@ export function MapPlan( {
...
 						stripZeros: true,
 					} ) }
-					/{ plan.plan_interval } { active_subscriptions }
+					/{ plan.plan_interval }
+					{ plan.active_subscriptions && (
+						<>
+							{ ' • ' }
+							{ sprintf(
+								// translators: %d is number of subscribers
+								_n( '%d active subscriber', '%d active subscribers', plan.active_subscriptions ),
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx]: `plan.active_subscriptions` is supposed to be boolean, so not sure what's going on here. Needs sorting out before sending string for translating.

https://github.com/Automattic/wp-calypso/blob/94899f83728b2f9d0717eb8d1fa7c298089b52a3/client/data/paid-newsletter/use-paid-newsletter-query.ts#L54-L63

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx [hunk 2]
```diff
@@ -64,11 +66,7 @@ export function MapPlan( {
 	tierToAdd,
 	selectedProductId,
 }: MapPlanProps ) {
-	let active_subscriptions = '';
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx:67]: Simplified by not storing string in a variable and moved inline into the component.

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx [hunk 3]
```diff
@@ -31,18 +33,18 @@ type MapPlanProps = {
 	selectedProductId: string;
 };
 
-function displayProduct( product?: Product ) {
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx:34]: Simplified by turning into a component which only outputs plan details, not "select" string.

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/connect-stripe.tsx [hunk 4]
```diff
@@ -55,8 +64,13 @@ export default function ConnectStripe( {
 					onClick={ () => {
 						recordTracksEvent( 'calypso_paid_importer_connect_stripe' );
 					} }
+					ariaLabel={ __( 'Connect Stripe' ) }
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/connect-stripe.tsx:67]: Added `ariaLabel`. I don't think previous implementation was screen reader friendly.

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx [hunk 5]
```diff
@@ -31,18 +33,18 @@ type MapPlanProps = {
...
-	}
+type ProductInfoProps = {
+	product: Product;
+};
 
+function ProductInfo( { product }: ProductInfoProps ) {
 	return (
-		<div>
-			<strong>{ product.title || '' }</strong>
+		<>
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx]: Simplified by removing `div` from DOM, but we'll need to check this doesn't break visually.

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx [hunk 6]
```diff
@@ -31,18 +33,18 @@ type MapPlanProps = {
...
+			{ product.title && <strong>{ product.title }</strong> }
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx]: Simplified by not having empty `<strong>` in DOM if title was empty.

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx (hunk 3 above)
Comment (simison) [reply client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx:34]: Removed the change from this PR.

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/success-notice.tsx [hunk 7]
```diff
@@ -1,9 +1,21 @@
...
+import { createInterpolateElement } from '@wordpress/element';
+import { sprintf } from '@wordpress/i18n';
 
 export default function SuccessNotice( { allEmailsCount }: { allEmailsCount: number } ) {
 	return (
 		<Notice status="success" className="importer__notice" isDismissible={ false }>
-			All set! We’ve found <strong>{ allEmailsCount } subscribers</strong> to import.
+			{ createInterpolateElement(
+				sprintf(
+					// @TODO: add _n() translation function once we finalize this copy.
```
Comment (simison) [inline client/my-sites/importer/newsletter/subscribers/paid-subscribers/success-notice.tsx:10]: TODO once we have final copy

File: client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx (hunk 1 above)
Comment (simison) [reply client/my-sites/importer/newsletter/subscribers/paid-subscribers/map-plan.tsx]: Removed the translation function for this string, we'll figure it outside this PR.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16901288

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @simison for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16901288).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16901288).

**Pull request merged**